core/Controller/LostController.php

```
@@ -259,19 +259,9 @@ public function setPassword($token, $userId, $password, $proceed) {
 	 * @throws \Exception
 	 */
 	protected function sendEmail($user) {
-		if (!$this->userManager->userExists($user)) {
-			throw new \Exception($this->l10n->t('Couldn\'t send reset email. Please make sure your username is correct.'));
-		}
-
 		$userObject = $this->userManager->get($user);
 		$email = $userObject->getEMailAddress();
```

Thanks for your pull request!

I didn't test it but I'm pretty sure that this line will now throw a exception if the user doesn't exists which might also indicate to the outside that the user doesn't exists.

I think we need to keep both checks, whether the user exists and whether we have a valid email address (see below). In both cases I would either throw a exception with a generic message, which doesn't indicate that the user (doesn't) exists, something like in L295 ("Couldn't send reset email. Please contact your administrator."). Or just return which will show the user a success message. As far as I know Mailman always return a success message. Don't know this is common sense and also used by most other web services. I'm fine with both, a generic error message or a generic success message.

@jancborchardt any preferences from your side?